examples/meta/src/meta_api/kwargs.sg

```diff
@@ -0,0 +1,38 @@
+# Keyword arguments are supported when constructing objects
+
+# Either when using factory functions:
+#Kernel k = kernel_machine(a=2)
+#Kernel k2 = kernel_machine("GaussianKernel", c=1.0)
+
+# Or when constructing objects from class constructors
+#GaussianKernel k3(1, a=2, b=3)
+
+# This fails (keyword arguments appear before normal arguments)
+#Kernel k4(asd=123,"SomeParam",kw=val)
+
+# This fails (keywords not allowed outside initialisations of variables)
+#k4 = kernel_machine("GaussianKernel", b=2)
+
+# This fails (keywords not allowed outside initialisation of variables)
+#Kernel k6 = kernel_machine("GaussianKernel", a=glob_fun(kw=123))
+
+# This is fine:
+#Kernel k6 = kernel_machine("GaussianKernel", a=glob_fun(ordinary_argument))
+
+# This also fails (keywords not allowed outside initialisation variables)
+#kernel_machine("GaussianKernel", a=glob_fun(kw=123))
+
+
+# REAL EXAMPLE
+# ------------
+# The cpp translation of the following line fails with exception:
+# > libc++abi.dylib: terminating with uncaught exception of type shogun::ShogunException
+# > [ERROR] In file /Users/esben/Code/shogun/src/shogun/base/class_list.h line 51: Type mismatchAbort trap: 6
+
+# GaussianKernel k0 = kernel_machine("GaussianKernel", log_width=2.0)
```

Comment:
Executing the C++ translation of this line, i.e.
```
auto k0 = kernel_machine("GaussianKernel");
k0->put("log_width", 2.0);
```
Throws the exception
```
libc++abi.dylib: terminating with uncaught exception of type shogun::ShogunException
[ERROR] In file /Users/esben/Code/shogun/src/shogun/base/class_list.h line 51: Type mismatchAbort trap: 6
```

@karlnapf can you guide me here? Don't know what I'm doing wrong.

I got the C++ interface working with the statements below this line (using class constructors), and I'm currently trying to get it working for the other interfaces.